packages/diagrams/frontend/sirius-components-diagrams/src/representation/DiagramSubscriptionProvider.tsx

```diff
@@ -61,15 +62,17 @@ export const DiagramSubscriptionProvider = memo(({ diagramId, editingContextId }
     <StoreContextProvider>
       <DialogContextProvider>
         <ImpactAnalysisDialogContextProvider>
-          <div
-            style={{ display: 'inline-block', position: 'relative' }}
-            data-representation-kind="diagram"
-            data-representation-label={state.diagramRefreshedEventPayload.diagram.metadata.label}>
-            <DiagramRenderer
-              key={state.diagramRefreshedEventPayload.diagram.id}
-              diagramRefreshedEventPayload={state.diagramRefreshedEventPayload}
-            />
-          </div>
+          <DiagramToolExecutorContextProvider>
```

Comment:
For the moment the context needs to be here because it needs to be under the `ImpactAnalysisDialogContextProvider`. This will probably change in a future PR if we manage to make the impact analysis dialog generic.